draft-kb-capsule-conversion.md

```diff
@@ -109,15 +111,15 @@ Upon receiving a Convertible Extended CONNECT Request, an HTTP intermediary MAY
 
 If the intermediary receives a correctly formed "101 (Switching Protocols)" response, it MUST change the response code to "200 (OK)".  If it receives a 2xx (Successful) response, it SHOULD return a "501 (Not Implemented)" status code, to indicate that the ":protocol" value was not accepted ({{!RFC9220, Section 3}}). Otherwise, it MUST forward any valid responses unmodified.  After sending a "200" response, the intermediary MUST process all further data to and from the server in accordance with the Capsule Protocol.
 
-## Converting an Extended CONNECT request to Extended CONNECT for a different HTTP version
+### Converting an Extended CONNECT request to Extended CONNECT for a different HTTP version
 
 An HTTP intermediary MAY translate a Convertible Extended CONNECT Request between different HTTP versions using ordinary HTTP version translation.
 
-## Adjusting for Server Capabilities
+## Requirements for Origin Servers
 
-When translating between HTTP versions, HTTP intermediaries often need to forward requests to servers that do not support all possible HTTP versions.  This can be accomplished by explicit configuration, by a fallback mechanism, or in some other way.
+In general, an origin server that supports multiple HTTP versions cannot predict which version will be used by an intermediary.  Unless otherwise constrained, the intermediary is free to use any HTTP version supported by the Origin Server, regardless of the HTTP version over which the intermediary received the request.  As a result, origin servers that intend to present a coherent view of their content must offer the same content on all supported HTTP versions.
 
-An HTTP intermediary implementing this specification MUST account for backend servers that support HTTP/2 or HTTP/3, but do not implement Extended CONNECT.  When the backend server does not support Extended CONNECT, the intermediary MUST forward the request over HTTP/1.1.
+When an intermediary supports this specification, the same rule applies to Capsule Protocol endpoints.  If an origin server intends to present a coherent view of its Capsule Protocol endpoints, and an intermediary that supports this specification might be in use, the origin server MUST offer the same endpoints across all supported HTTP versions.  Accordingly, the origin server MUST enable Extended CONNECT if it supports HTTP/2 or HTTP/3.
```

Comment:
While the preceding paragraph might be true, I'm not sure if RFC 9110 has something equivalent to the MUSTs in this paragraph.

My gut feeling is that our requirements for cross-version behavior should be identical to that of ordinary HTTP, because extended connect runs on top of HTTP.

All I know is that [RFC 9110 Section 6.2](https://www.rfc-editor.org/rfc/rfc9110.html#name-control-data) provides recommendations (but not requirements) for how the clients select versions, while it is almost silent regarding the requirements for servers.

Could we just point to that section and call it a day?

Reply:
In my view, the "HTTP/2 without Extended CONNECT" situation is a real mess that should be highlighted somehow.

I've tried a different approach in #4.  Please review.